presto-clp/src/main/java/com/facebook/presto/plugin/clp/ClpMetadataFilterProvider.java

```diff
@@ -0,0 +1,264 @@
+ *
+ *   <li><b>{@code rangeMapping}</b> (optional): remaps a logical filter to physical metadata-only columns.
+ *       This field is valid only for numeric-type columns.
+ *       For example, a condition such as:
+ *       <pre>{@code
+ *       "msg.timestamp" > 1234 AND "msg.timestamp" < 5678
+ *       }</pre>
+ *       will be rewritten as:
+ *       <pre>{@code
+ *       "end_timestamp" > 1234 AND "begin_timestamp" < 5678
+ *       }</pre>
+ *       This ensures the filter applies to a superset of the actual result set, enabling safe pruning.</li>
+ *
+ *   <li><b>{@code required}</b> (optional, default: {@code false}): indicates whether the filter must be present
+ *       in the extracted metadata filter SQL query. If a required filter is missing or cannot be pushed down,
+ *       the query will be rejected.</li>
+ * </ul>
+ */
+public class ClpMetadataFilterProvider
+{
+    private final Map<String, List<Filter>> filterMap;
+
+    @Inject
+    public ClpMetadataFilterProvider(ClpConfig config)
+    {
+        requireNonNull(config, "config is null");
+
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            filterMap = mapper.readValue(
+                    new File(config.getMetadataFilterConfig()),
+                    new TypeReference<Map<String, List<Filter>>>() {});
+        }
+        catch (IOException e) {
+            throw new PrestoException(CLP_METADATA_FILTER_CONFIG_NOT_FOUND, "Failed to metadata filter config file open.");
+        }
+    }
+
+    public void checkContainsRequiredFilters(SchemaTableName schemaTableName, String metadataFilterSql)
+    {
+        boolean hasRequiredMetadataFilterColumns = true;
+        ImmutableList.Builder<String> notFoundListBuilder = ImmutableList.builder();
+        for (String columnName : getRequiredColumnNames(format("%s.%s", CONNECTOR_NAME, schemaTableName))) {
+            if (!metadataFilterSql.contains(columnName)) {
+                hasRequiredMetadataFilterColumns = false;
+                notFoundListBuilder.add(columnName);
+            }
+        }
+        if (!hasRequiredMetadataFilterColumns) {
+            throw new PrestoException(
+                    CLP_MANDATORY_METADATA_FILTER_NOT_VALID,
+                    notFoundListBuilder.build() + " is a mandatory metadata filter column but not valid");
+        }
+    }
+
+    /**
+     * Rewrites the input SQL string by remapping filter conditions based on the configured
+     * metadata filter range mappings for the given scope.
+     *
+     * <p>The {@code scope} follows the format {@code catalog[.schema][.table]}, and determines
+     * which filter mappings to apply. For each level of scope (catalog, schema, table), this
+     * method collects all range mappings defined in the metadata filter configuration. Mappings
+     * from more specific scopes (e.g., table-level) override or supplement those from broader
+     * scopes (e.g., catalog-level).
```

Comment:
```suggestion
     * which filter mappings to apply, since mappings from more specific scopes (e.g., table-level)
     * override or supplement those from broader scopes (e.g., catalog-level). For each scope
     * (catalog, schema, table), this method collects all range mappings defined in the metadata
     * filter configuration.
```